src/com/facebook/buck/testrunner/TestNGRunner.java

```diff
@@ -114,6 +131,30 @@ private boolean shouldIncludeTest(String className) {
     return testSelectorList.isIncluded(testDescription);
   }
 
+  private static String calculateTestMethodName(ITestResult iTestResult) {
+    Object[] parameters = iTestResult.getParameters();
+    String name = iTestResult.getName();
+
+    if (parameters == null || parameters.length == 0) {
+      return name;
+    }
+
+    StringBuilder builder = new StringBuilder(name).append(" (");
+    builder.append(
+        Arrays.stream(parameters)
+            .map(
+                parameter -> {
+                  try {
+                    return parameter == null ? "null" : parameter.toString();
+                  } catch (Exception e) {
+                    return "Unstringable object";
```

Comment:
Why do we need to catch it and not fail fast?

Reply:
we can fail instead, but I don't think "failure to `toString` an object" should be enough to block a developer from seeing their test results